### ISSUE TITLE
🐛 [Bug]

### ISSUE THREAD

**Issue by ddk2001** [Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8#225]
## 🐛 Bug report
Found a BUG, in the copy of Blackstone Abyss, boss's creature_template entry： 9018. When the blood hit about 30%, it appeared: unable to attack in escape state. Then the server crashed. At the same time, there are still many tasks that cannot be completed in dialogue with NPC.
![TEK`(6F849G_~_T 8RR6K4O](https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/assets/29648605/fcd7e94c-e969-4df5-b775-6337e9336f14)

- No information

### Expected behavior

- No information

### Steps to reproduce

1.
2.
3.
4. ...
5. Profit.

### Version

Client Version: 
- No information

**Comment by leelf00** [comment]
Ok,I will debug it.

**Comment by ddk2001** [comment]
Later, I researched it and found out that the problem was that as long as the character was charmed, ran away in fear, transformed into a small animal, etc., the server would go down.

**Comment by ddk2001** [comment]
Once this quest has started, horseback riding will cause the server to go down. Several places have been found to cause server crashes. Basically, it is down because the character is not controlled.
![MATNJ({KE@6D~Y4$}T0MCK5](https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/assets/29648605/f216d208-7c97-4d46-8956-48fa94b8ef5d)

**Comment by leelf00** [comment]
quest id?

**Comment by ddk2001** [comment]
14293

**Comment by leelf00** [comment]
Can you post crash log?

**Comment by ddk2001** [comment]
I didn't compile the debug. I'll compile a copy today.

**Comment by ddk2001** [comment]
![EN_FQ@26WC_5IR(L_M0363T](https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/assets/29648605/74e70c13-1a93-4af7-8aad-15188a3f3bc1)
![XQ(GQC@0XB0MZRGK4Z_XYJD](https://github.com/Legends-of-Azeroth/Legends-of-Azeroth-Pandaria-5.4.8/assets/29648605/5d8305a6-4f0a-40f3-95d3-5b592247e487)
When I log in in DeBug mode, this is prompted.

**Comment by leelf00** [comment]
Any crash logs from logs folder?

**Comment by ddk2001** [comment]
the release is  not  Any crash logs from logs folder。can you test it？

**Comment by ddk2001** [comment]
I'm going to check this debug log on the office computer on Monday.

**Comment by ddk2001** [comment]
Application Log
Start: Tue Apr  2 10:05:54 2024


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: (low.x <= high.x) && (low.y <= high.y) && (low.z <= high.z)
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:97


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: ! low.isNaN() && ! high.isNaN()
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:98


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: (low.x <= high.x) && (low.y <= high.y) && (low.z <= high.z)
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:97


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: ! low.isNaN() && ! high.isNaN()
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:98


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: (low.x <= high.x) && (low.y <= high.y) && (low.z <= high.z)
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:97


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: ! low.isNaN() && ! high.isNaN()
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:98


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: (low.x <= high.x) && (low.y <= high.y) && (low.z <= high.z)
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:97


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: ! low.isNaN() && ! high.isNaN()
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:98


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: (low.x <= high.x) && (low.y <= high.y) && (low.z <= high.z)
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:97


**************************

Assertion Failure - worldserver.exe
Debug assertion failure

Expression: ! low.isNaN() && ! high.isNaN()
G:\yuanma\4.1ceshi\Legends-of-Azeroth-Pandaria-5.4.8\dep\g3dlite\include\G3D/AABox.h:98

**Comment by ddk2001** [comment]
There is only so much for this log file to be recorded. The following error message is not logged.

**Comment by leelf00** [comment]
Seems gameobject issue

**Comment by ddk2001** [comment]
I think it's also a matter of roles. Once the play is controlled, the server goes down.

**Comment by ddk2001** [comment]
Did you test it later? Have you found the reason for the downtime? I took a look at gameobject.cpp. I didn't see anything wrong.

**Comment by ddk2001** [comment]
I found that your source code has the same bug as pandaria, when you buy more than 200,000 gold coins at one time, the gold will not be deducted.

**Comment by leelf00** [comment]
Yes.We fork from alexkulya's source and have most same bugs.

**Comment by leelf00** [comment]
How to reproduce? use battle coin buy gold coins?

**Comment by ddk2001** [comment]
No, for example, if I buy 2,200 gold coins at one time, the total amount of gold required is 220,000 gold. However, no gold will be deducted. I think it should be an issue with int32 limitations

**Comment by ddk2001** [comment]
Buy-ins above 2147483647 will not be deducted for gold. This is exactly the upper limit of 32 bits

**Comment by leelf00** [comment]
I push a commit due to money limit. Try last commit

**Comment by ddk2001** [comment]
How did you find it. The way I came up with was to change the upper limit of the MAX_MONEY_AMOUNT

**Comment by leelf00** [comment]
MAX_MONEY_AMOUNT is uint64.

**Comment by leelf00** [comment]
The 32bit uint32 limit is caused by code. DB column is also right.

**Comment by ddk2001** [comment]
You just have to find the specific location.

**Comment by ddk2001** [comment]
For advice, this version of the item upgrade is only up to level 4, and I remember that the official uniform has level 6 or 8. Do you know how to modify the location.

**Comment by leelf00** [comment]
Could you find some wowhead or old wiki links for me?

**Comment by ddk2001** [comment]
I have a wiki here, but I don't think it's very useful.  https://warcraft.huijiwiki.com/wiki/%E9%A6%96%E9%A1%B5

**Comment by leelf00** [comment]
https://wowpedia.fandom.com/wiki/Patch_5.4.8    
In MOP, item can upgrade 2 times.

**Comment by ddk2001** [comment]
I think this source code should have been modified. But I can't find the location where this item was upgraded, and the database hasn't found it either. I think it should be in DBC.

**Comment by qyh214** [comment]
> For advice, this version of the item upgrade is only up to level 4, and I remember that the official uniform has level 6 or 8. Do you know how to modify the location.

As far as I know, the servers in Chinese Mainland were special at that time, and the number of upgrades was different from that in other regions
